pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -905,6 +905,8 @@ func (qjm *XController) getAggregatedAvailableResourcesPriority(unallocatedClust
 			continue
 		} else if !value.Status.CanRun {
 			klog.V(11).Infof("[getAggAvaiResPri] %s: Skipping adjustments for %s since it can not run.", time.Now().String(), value.Name)
+			totalResource := qjm.addTotalSnapshotResourcesConsumedByAw(value.Status.TotalGPU, value.Status.TotalCPU, value.Status.TotalMemory)
+			preemptable = preemptable.Add(totalResource)
 			continue
 		} else if value.Status.SystemPriority < targetpr {
 			// Dispatcher Mode: Ensure this job is part of the target cluster
```

Comment:
I am confused by this case (`value.Status.SystemPriority < targetpr`). Shouldn't the `preemptableAWs` computation happen in both agent and dispatch mode (for the target cluster). It looks to me like the curly at line 930 should be moved up to line 926 moving the 927-929 block past the`if qjm.Dispatcher` test. This code is not novel to the PR and I am not sure I fully understand. Bottom line is we should have look at this, possibly in a separate PR.

The agent mode is missing a log message (which the curly change above would fix).

Reply:
AW that is preemptable it's canRun is false. `else if` loop above this will catch such AWs and add it to preemptable accounting

Reply:
To add not doing any changes for dispatcher mode due to testing constraints

Reply:
With the current implementation `preemptableAWs` is always empty in Agent mode. I don't think that is correct. But this is not a new issue. So I agree we can move the diccussion to a separate issue/PR.

Please add a log message to track the case `if value.Status.SystemPriority < targetpr {` line 934.

Reply:
Thanks, Done.